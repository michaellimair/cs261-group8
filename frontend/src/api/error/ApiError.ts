class ApiError<T> extends Error {
  data?: T;

  constructor(message: string, data?: T) {
    super(message ?? 'An error has occurred in the API.');
    this.data = data;
  }

  static isApiError(obj: any): obj is ApiError<any> {
    return obj instanceof ApiError;
  }
}

export default ApiError;
