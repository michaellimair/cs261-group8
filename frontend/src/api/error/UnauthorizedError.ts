import ApiError from './ApiError';

class UnauthorizedError extends ApiError<null> {
  constructor(message?: string) {
    super(message ?? 'Unauthorized', null);
  }

  static isUnauthorizedError(obj: any): obj is UnauthorizedError {
    return obj instanceof UnauthorizedError;
  }
}

export default UnauthorizedError;
