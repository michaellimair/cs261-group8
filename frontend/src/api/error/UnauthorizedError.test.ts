import ApiError from './ApiError';
import UnauthorizedError from './UnauthorizedError';

describe('UnauthorizedError', () => {
  it('is instantiated successfully', () => {
    const errorData = {
      status_code: 400,
      message: 'An error has occurred',
      errors: [],
    };

    const err = new UnauthorizedError(errorData.message);

    expect(err.message).toBe(errorData.message);
    expect(err).toMatchSnapshot();
  });

  it('is a subclass of ApiError', () => {
    const err = new UnauthorizedError('error');

    expect(ApiError.isApiError(err)).toBe(true);
  });

  describe('isUnauthorizedError', () => {
    it('identifies UnauthorizedError classes correctly', () => {
      const err = new UnauthorizedError('error');

      expect(UnauthorizedError.isUnauthorizedError(err)).toBe(true);
    });
  });
});
