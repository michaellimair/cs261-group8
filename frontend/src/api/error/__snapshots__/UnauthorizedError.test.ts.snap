// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthorizedError is instantiated successfully 1`] = `[Error: An error has occurred]`;
